### DISCUSSION TITLE
How to disable/remove an event?

### DISCUSSION THREAD

**Discussion by silveltman** [alpinejs/alpine#2626] [1. Help]
I want to remove/disable an event to increase performance.

Basically, it's a parrallax and I only want the event to fire when neccesary, so when the element is in the viewport. Is this possible with alpine?

```
<div
x-data="{ 
  handleScroll() {
    let value = window.pageYOffset / 3
    $refs.img.style.transform = `translateY( ${value}px )`
  }
}"
@scroll.window="handleScroll"
>
  <img x-ref="img" src="my/image" alt="my-alt">
  <div>
    <h1>Lorem</h1>
    <p>Lorem ipsum</p>
  </div>
</div>
```

**Comment by SimoTod** [comment]
You can shortcuit the event with a flag and use the intersect plugin to toggle on and off but you don't need js at all for a parallax effect. A pure css implementation is more performant and it won't cause any repainting when the div is outside the viewport. If you google 'parallax pure css' you should be able to find many examples.

**Reply by silveltman** [reply]
Thanks for the tip!

For the people landing on this topic, this video's explains it really well: https://www.youtube.com/watch?v=mxHoPYFsTuk&t=543s
